src/pudl/transform/ferc1.py

```diff
@@ -136,30 +139,8 @@ def params(self) -> Ferc1TableTransformParams:
         """Obtain table transform parameters based on the table ID."""
         return Ferc1TableTransformParams.from_id(table_id=self.table_id)
 
-    @abstractmethod
-    def transform(
-        self,
-        raw_dbf: pd.DataFrame,
-        raw_xbrl_instant: pd.DataFrame,
-        raw_xbrl_duration: pd.DataFrame,
-    ) -> pd.DataFrame:
-        """Abstract FERC Form 1 specific transformation method.
-
-        This method primarily exists to define the FERC 1 specific call signature.
-
-        Params:
-            raw_xbrl_instant: Table representing raw instantaneous XBRL facts.
-            raw_xbrl_duration: Table representing raw duration XBRL facts.
-            raw_dbf: Raw Visual FoxPro database table.
-
-        Returns:
-            A single transformed table concatenating multiple years of cleaned data
-            derived from the raw DBF and/or XBRL inputs.
-
-        """
-        ...
-
-    def concat_dbf_xbrl(
+    @cache_df(key="main")
```

Comment:
Here `key` is literally the key used to store the dataframe in the internal dictionary. It can be any string. The more different `key`s are specified, the more intermediate dataframes are going to get stored. If you use the same `key` more than once then yes, the dataframe gets overwritten.

They way I'm using it here it's an incrementally updating record of the last state the dataframe had when passed through a decorated function successfully. If need be, you could give every step its own distinct key and save them all separately, but that would use a lot of memory.  It might be useful in the midst of development when you're trying to figure out what step a problem is happening in though.

I kept separate `xbrl` and `dbf` dataframes since they have their own origins and go through different paths before merging into the main transform.